okta/services/idaas/resource_okta_network_zone.go

```diff
@@ -302,6 +334,14 @@ func validateNetworkZone(d *schema.ResourceData) error {
 	if d.Get("usage").(string) != "POLICY" && ok && proxies.(*schema.Set).Len() != 0 {
 		return fmt.Errorf(`zones with usage = "BLOCKLIST" cannot have trusted proxies`)
 	}
+
+	// Validate that use_as_exempt_list is only used with IP zones
```

Comment:
We might be able to put this logic into a schema validator,
This would mean that things can fail earlier during the `terraform plan` phase